Написаны тесты по POM-модели

lnlydrvr opened the pull request.

lnlydrvr committed:
Написаны тесты по POM-модели

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,13 @@
+
+@pytest.fixture(scope='function')
+def driver():
+    service = FirefoxService(executable_path="C:\Installed\Soft\WebDriverFirefox\geckodriver.exe")
```
Нужно исправить: использовать executable_path нельзя, не заведется в ci

lnlydrvr replied:
Исправлено в коммите [1660ac5](https://github.com/lnlydrvr/Sprint_6/pull/1/commits/1660ac57dbea2f46485c266fcb32130f0c77cebe) с применением webdriver-manager.

AlexKitIt commented on pages/default_page.py:
```diff
@@ -0,0 +1,40 @@
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.wait import WebDriverWait
+
+class DefaultPage:
```
Стоит называть этот класс BasePage или BaseView. POM тут четко это описывает

lnlydrvr replied:
Исправлено в коммите [1660ac5](https://github.com/lnlydrvr/Sprint_6/pull/1/commits/1660ac57dbea2f46485c266fcb32130f0c77cebe).

AlexKitIt commented on tests/test_order_page.py:
```diff
@@ -0,0 +1,46 @@
+    def test_make_order_from_header(self, driver, user_info, rent_info):
+        MainPage(driver).open_user_info_form_by_header_button()
+
+        OrderPage(driver).wait_user_info_form()
```
Нужно исправить: здесь и далее: объект пейджи создается один раз, записывается в переменную и потом используется. каждый отдельный объект - это дополнительное место в памяти

lnlydrvr replied:
Исправлено в коммите [1660ac5](https://github.com/lnlydrvr/Sprint_6/pull/1/commits/1660ac57dbea2f46485c266fcb32130f0c77cebe).

lnlydrvr committed:
Убран файл к драйверу в conftest.py, переименован
класс в BasePage, сделаны переменные для page для
оптимизации использования памяти
